### PR TITLE
DNN-10015 :  <source> tag added to filtered tags.

### PR DESCRIPTION
**Pull request by aliazizoglu**

### CODE REVIEW

File: DNN Platform/Tests/DotNetNuke.Tests.Core/Security/PortalSecurity/PortalSecurityTest.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            //Arrange
+            var portalSecurity = new DotNetNuke.Security.PortalSecurity();
+            var htmlEntry = @"<source>
+                               Hi this is personal site
+                               Test for source tag  
+                              </source> ";
+            var markup = DotNetNuke.Security.PortalSecurity.FilterFlag.NoMarkup |
+                         DotNetNuke.Security.PortalSecurity.FilterFlag.NoScripting |
+                         DotNetNuke.Security.PortalSecurity.FilterFlag.NoAngleBrackets;
+            var expectedOutput = "  ";
```
Comment (galatrash-at-dnn) [inline DNN Platform/Tests/DotNetNuke.Tests.Core/Security/PortalSecurity/PortalSecurityTest.cs]: IMO, we need to have 3 or more separate tests for various combinations settings for these flags:
- none of these flags set; should return the same without filtering
- NoMarkup
- NoScripting
- NoAngleBrackets
- combination of 2 of these together (3 cases)
- all of them together (which is this test).

File: DNN Platform/Tests/DotNetNuke.Tests.Core/Security/PortalSecurity/PortalSecurityTest.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        #region Tear Down
+        [TearDown]
+        public void TearDown()
+        {
+            //TODO:implement clean up
+        }
+        #endregion
+
+        #region Test Methods
+        [Test]
```
Comment (vano-maisuradze) [inline DNN Platform/Tests/DotNetNuke.Tests.Core/Security/PortalSecurity/PortalSecurityTest.cs]: We can use **TestCase** to easily test different scenarios. Like this:
```
[TestCase("<source></source>", "")]
[TestCase("<source>Hi this is personal Test for source tag</source>   ", "   ")]
[TestCase("<SOURCE>Upper case</SOURCE>", "")]
[TestCase("<source src="https://google.com/">Source with attribute</source>", "")]
[Add more test cases here]
public void Typing_Html_Source_Tag_Should_Not_Be_Allowed(string html, string expectedOutput)
{
        // Use html parameter as input and expectedOutput to assert the result
}
```

Comment (vano-maisuradze) [reply DNN Platform/Tests/DotNetNuke.Tests.Core/Security/PortalSecurity/PortalSecurityTest.cs]: OMHO, **Typing_** prefix is not necessary in the name. **Html_Source_Tag_Should_Not_Be_Allowed** is better.

**Comment by vano-maisuradze** [comment]
All tests are passing. Merging to `development` branch

**Pull request merged**